### PR TITLE
[20542] Improve CI version management

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR unifies the CI in all workflows to follow this behavior:
- Trigger CI in all pull requests which HEAD branch is set to this supported branch version
- Remove any CI trigger on push events
- Run nightly events in all supported branches as a matrix.
- Include missing ignore files for the sanitizer event trigger.
- Rename sanitizer job as "address-sanitizer"

The following PRs are the counterpart in the supported branches:
- **2.13.x**: #4481
- **2.12.x** :#4482
- **2.10.x** :#4483
- **2.6.x**: #4484

The follwing backports have been modified accoddingly:
 - **2.13.x**: #4477
- **2.12.x** :#4473
- **2.10.x** :#4475
- **2.6.x**: #4476

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Changes are ABI compatible. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by elianalf — DISMISSED** [review]

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

File: .github/workflows/nightly-mac-ci.yml [hunk 1]
```diff
@@ -7,17 +7,33 @@ on:
...
+          - '2.10.x'
+          - '2.6.x'
     uses: ./.github/workflows/reusable-mac-ci.yml
     with:
-      label: 'nightly-sec-mac-ci'
+      label: 'nightly-sec-mac-ci-${{ matrix.fastdds-branch }}'
       cmake-args: "-DSECURITY=ON"
       ctest-args: "-LE xfail"
-      fastdds-branch: 'master'
+      fastdds-branch: '${{ matrix.fastdds-branch }}'
```
Comment (elianalf) [inline .github/workflows/nightly-mac-ci.yml:23]:
```suggestion
      fastdds-branch: ${{ matrix.fastdds-branch }}
```

**Review by elianalf — DISMISSED** [review]

**Review by elianalf — APPROVED** [review]

File: .github/workflows/documentation-tests.yaml
Comment (EduPonz) [inline .github/workflows/documentation-tests.yaml:1]: Remove push event here please

File: .github/workflows/sanitizer-tests.yaml
Comment (EduPonz) [inline .github/workflows/sanitizer-tests.yaml:1]: Rename to address-sanitizers

File: .github/workflows/thread-sanitizer.yaml
Comment (EduPonz) [inline .github/workflows/thread-sanitizer.yaml:1]: Add cron here

**Review by EduPonz — CHANGES_REQUESTED** [review]
As agreed, please remove all `push` events

**Review by EduPonz — DISMISSED** [review]

**Review by EduPonz — APPROVED** [review]

**Pull request merged**